libs/nest/core/src/lib/decorators/ParsedUserAgent.decorator.ts

```diff
@@ -0,0 +1,92 @@
+import { getRequest } from '@island.is/auth-nest-tools'
+import { createParamDecorator, ExecutionContext } from '@nestjs/common'
+
+// Old iOS app user agents:
+// - IslandApp/144 CFNetwork/1498.700.2 Darwin/23.6.0
+
+// Old Android app user agents. We associate it with IslandIsApp, assuming
+// there are no other React Native android apps calling our API.
+// - okhttp/4.9.2
+
+// New app user agents
+// - IslandIsApp (1.0.0}) Build/321 (ios/9.0.0)
+// - IslandIsApp (1.0.0}) Build/321 (android/9.0.0)
+
+export interface UserAgent {
+  ua: string
+  os: {
+    name?: 'iOS' | 'Android'
+    version?: string
+  }
+  app: {
+    name?: 'IslandIsApp'
+    version?: string
+    build?: number
+  }
+}
+
+/**
+ * Parses the user agent string and returns an object with the user agent, os and app information.
+ *
+ * Only supports parsing IslandIsApp user agents to start with but the interface is
+ * future compatible if we want to parse more user agents with ua-parser-js.
+ */
+export const parseUserAgent = (userAgentString: string): UserAgent => {
```

Comment:
A micro-optimization could be to move the regex definitions outside the function to avoid having to recompile it each time, since this can get called quite often?

Reply:
Inline regex patterns are compiled once when loading the script according to spec. The only thing which could be saved is object allocation and GC, which is negligent imo.